src/feature/control/control_getinfo.c

```diff
@@ -1282,6 +1283,9 @@ getinfo_helper_events(control_connection_t *control_conn,
       *answer = descs;
       routerinfo_free(r);
       extrainfo_free(e);
+    } else if (!strcmp(question, "status/dos-mitigation")) {
+      *answer = tor_strdup(dos_getinfo_helper_format_stats());
```

Comment:
This code leaks the newly allocated string from dos_getinfo_helper_format_stats().
```suggestion
      *answer = dos_getinfo_helper_format_stats();
```